packages/common/src/lib/common/tools.ts

```diff
@@ -1382,19 +1425,18 @@ function getAdapterDir(adapter) {
 /**
  * Returns the hostname of this host
  * @alias getHostName
- * @returns {string}
  */
-function getHostName() {
+function getHostName(): string {
     // for tests purposes
     if (process.env.IOB_HOSTNAME) {
         return process.env.IOB_HOSTNAME;
     }
     try {
         const configName = getConfigFileName();
         const config = fs.readJSONSync(configName);
-        return config.system ? config.system.hostname || require('os').hostname() : require('os').hostname();
+        return config.system ? config.system.hostname || os.hostname() : os.hostname();
```

Comment:
```suggestion
        return config.system?.hostname || os.hostname();
```
(or `??` instead of `||`) if we know that hostname is never empty.